dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/DataverseSaverIT.java

```diff
@@ -128,6 +157,25 @@ public void saveEditedDataverse_WithNonExistingDataverse() {
 
     }
 
+    private void cleanupDatabase() throws NotSupportedException, SystemException, RollbackException, HeuristicMixedException, HeuristicRollbackException {
+        transaction.begin();
+
+        dropAllTables();
```

Comment:
Why do we have to drop the whole database and then rebuild it here? Isn't ROLLBACK enough?

Reply:
The rollback is not executed above the method that was fixed, you can't really do a rollback above that method, so you have to cleanup the database manually.

Reply:
Still dropping whole database seems like a harsh treatment. Wouldn't it be possible to rollback manually transaction and then manually remove only the data connected to mail sending?